build.rs

```diff
@@ -1,22 +1,62 @@
 use std::env;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
+
+fn path_exists(path: &str) -> bool {
+    Path::new(path).exists()
+}
+
+fn find_julia() -> Option<String> {
+    if let Ok(path) = env::var("JL_PATH") {
+        return Some(path);
+    }
+
+    if path_exists("/usr/include/julia/julia.h") {
+        return Some("/usr".to_string());
+    }
+
+    let include_path = env::var("CPATH")
+        .or_else(|_| env::var("C_INCLUDE_PATH"));
+
+    if let Ok(paths) = include_path {
+        for path in paths.split(":") {
+            let mut buf = PathBuf::from(path);
+            buf.push("julia.h");
+            if buf.exists() {
+                // Clang already knows how to find julia in this case.
+                return None;
+            }
+        }
+    }
+
+    panic!("Unable to find julia installation. Please install julia or provide the installation path as the JL_PATH environment variable.")
+}
 
 fn main() {
-    let flags = match env::var("JL_PATH") {
-        Ok(path) => {
+    let mut out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
+    out_path.push("bindings.rs");
+
+    println!("cargo:rerun-if-changed=wrapper.h");
+    println!("cargo:rerun-if-env-changed=JL_PATH");
+
+    if let Ok(_) = env::var("CARGO_FEATURE_DOCS_RS") {
```

Comment:
This works because of the stuff you've added to Cargo.toml, right? (I tested it by running docs.rs locally and it really seems to do the trick :)).

Reply:
Yep. The `docs-rs` feature is not a special feature name, but you can configure `docs.rs` to compile the crate with certain features enabled. You can also compile with
```
cargo doc --features docs-rs
```
to avoid having to run docs.rs locally.

Reply:
Ah, TIL!